### PR TITLE
#547/create stats page with chart

### LINKED ISSUE #547: ⚛️ Stats page
We have a history page. It's time to stack a page with more stats, using step line chats to report the level history and pie charts to report the win/loss ratio

### PR DESCRIPTION
**Pull request by Damrod**
closes #547

### CODE REVIEW

File: webapp/src/pages/UserMatchHistoryPage/UserMatchHistoryPage.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { ButtonVariant, IconVariant, RowItem } from '../../shared/components';
```
Comment (Damrod) [inline webapp/src/pages/UserMatchHistoryPage/UserMatchHistoryPage.tsx:1]: this one is actually a renamed UserStatsPage

File: webapp/package.json [hunk 2]
```diff
@@ -4,15 +4,19 @@
...
     "react": "^18.2.0",
+    "react-chartjs-2": "^5.2.0",
     "react-confirm-alert": "^3.0.6",
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.4.5",
     "react-scripts": "^5.0.1",
     "socket.io-client": "^4.5.4",
-    "web-vitals": "^3.1.0",
-    "transcendence-shared": "^1.0.0"
+    "transcendence-shared": "^1.0.0",
```
Comment (Damrod) [inline webapp/package.json:18]: I think npm automatically makes this in alphabetic order

File: webapp/src/pages/UserStatsPage/UserStatsPage.tsx [hunk 3]
```diff
@@ -1,89 +1,71 @@
...
-        url: user ? `${AVATAR_EP_URL}/${user.avatarId}` : '',
-        XCoordinate: user ? user.avatarX : 0,
-        YCoordinate: user ? user.avatarY : 0,
-      }}
-      avatarLabel={username!}
-      avatarCaption={`level ${user?.level ?? 1}`}
-      avatarLinkTo={`${USER_URL}/${username!}`}
-    />
+    <div className="stats-page">
+      <Line
```
Comment (Damrod) [inline webapp/src/pages/UserStatsPage/UserStatsPage.tsx:43]: This is fun

File: webapp/src/pages/UserPage/UserPage.tsx [hunk 4]
```diff
@@ -194,17 +200,13 @@ export default function UserPage() {
                 <Button
                   {...{
                     variant: ButtonVariant.SUBMIT,
-                    iconVariant: IconVariant.PLAY,
+                    iconVariant: IconVariant.STATS,
```
Comment (Damrod) [inline webapp/src/pages/UserPage/UserPage.tsx:203]: swapped the buttons, picked better colors, and added stats icon from figma

File: webapp/src/pages/UserStatsPage/UserStatsPage.tsx [hunk 5]
```diff
@@ -1,89 +1,115 @@
...
-    />
+    <div className="stats-page">
+      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        {`game stats for '${user.username}'`}
+      </Header>
+      <div className="graphs">
+        <div className="line-graph-wrapper">
+          <Line
+            data={{
+              labels: data.map((item) => item.timestamp),
```
Comment (Damrod) [inline webapp/src/pages/UserStatsPage/UserStatsPage.tsx:64]: we could customize these colors, but tbh, I like the default ones, so I'm not changing them.

File: webapp/src/pages/UserStatsPage/UserStatsPage.css [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  display: flex;
+  justify-content: center;
+}
+
+.stats-page .graphs .line-graph-wrapper canvas {
+  min-height: 18rem;
+  max-width: 60vw;
+}
+
+.stats-page .graphs .pie-chart-wrapper {
```
Comment (Damrod) [inline webapp/src/pages/UserStatsPage/UserStatsPage.css:39]: the style inside the canvas responds to css changes only on reload, no hot reloading. I.e. The component will create element.style based on the style sheet and only update it on reload.

File: webapp/src/pages/UserStatsPage/UserStatsPage.css [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  gap: var(--row-gap) var(--column-gap);
+}
+
+.stats-page .graphs .line-graph-wrapper {
+  display: flex;
+  justify-content: center;
+}
+
+.stats-page .graphs .line-graph-wrapper canvas {
+  min-height: 18rem;
```
Comment (Damrod) [inline webapp/src/pages/UserStatsPage/UserStatsPage.css:35]: the style could maybe use some more love, but it looks good, it's boring and I'm tired, so this is it.

File: webapp/src/pages/UserMatchHistoryPage/UserMatchHistoryPage.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      isNotFound={!user}
+      fetchFn={getUserGames}
+      dataMapper={mapGameToRow}
+      avatarProps={{
+        url: user ? `${AVATAR_EP_URL}/${user.avatarId}` : '',
+        XCoordinate: user ? user.avatarX : 0,
+        YCoordinate: user ? user.avatarY : 0,
+      }}
+      button={{
+        variant: ButtonVariant.SUBMIT,
```
Comment (Damrod) [inline webapp/src/pages/UserMatchHistoryPage/UserMatchHistoryPage.tsx:87]: With an added button here

**Review by gbudau — APPROVED** [review]
Thank you for this feature! The chart looks great :tada:

**Pull request merged**